File: 0978-longest-turbulent-subarray/0978-longest-turbulent-subarray.py
class Solution:
    def maxTurbulenceSize(self, arr: List[int]) -> int:
        sign = ""
        ans = count = 1
        for i, n in enumerate(arr[1:]):
            if sign == "<":
                if n < arr[i]:
                    count += 1
                    sign = ">"
                else:
                    ans = max(ans, count)
                    if n > arr[i]:
                        count = 2
                    else:
                        count = 1
                        sign = ""
            elif sign == ">":
                if n > arr[i]:
                    count += 1
                    sign = "<"
                else:
                    ans = max(ans, count)
                    if n < arr[i]:
                        count = 2
                    else:
                        count = 1
                        sign = ""
            else:
                count = 2
                if n > arr[i]:
                    sign = "<"
                elif n < arr[i]:
                    sign = ">"
                else:
                    count = 1
                    sign = ""
        return max(ans, count)

File: 0978-longest-turbulent-subarray/0978-longest-turbulent-subarray.cpp
class Solution {
public:
    int maxTurbulenceSize(vector<int>& arr) {
        char sign = '=';
        int count = 1, ans = 1, N = arr.size() - 1;
        for (int i = 0; i < N; ++i){
            if (sign == '<'){
                if(arr[i] < arr[i+1]){
                    ++count;
                    sign = '>';
                }
                else{
                    ans = max(ans, count);
                    count = 1;
                    if (arr[i] == arr[i+1]) sign = '=';
                    else count += 1;
                }
            }
            else if (sign == '>'){
                if(arr[i] > arr[i+1]){
                    ++count;
                    sign = '<';
                }
                else{
                    ans = max(ans, count);
                    count = 1;
                    if(arr[i] == arr[i+1]) sign = '=';
                    else count += 1;
                }
            }
            else {
                count = 2;
                if(arr[i] < arr[i+1]){
                    sign = '>';
                }
                else if (arr[i] > arr[i+1]){
                    sign = '<';
                }
                else{
                    sign = '=';
                    count = 1;
                }
            }
        }
        return max(ans, count);
    }
};